EIPS/eip-medgasprice_opcode.md

```diff
@@ -0,0 +1,93 @@
+---
+eip: <to be assigned>
+title: OPCODE 0x46 MEDGASPRICE
+author: Justice Hudson (@jchancehud)
+discussions-to: https://ethereum-magicians.org/t/medgasprice-opcode-eip/5480
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-05
+---
+
+## Simple Summary
+
+An opcode allowing smart contracts to access information about recent gas prices.
+
+## Abstract
+
+There is an existing opcode `0x3a GASPRICE` that allows contracts to access the gas price of the current transaction. This value can be used to impose a maximum gas price on contract calls by setting a constant value, but cannot be used as a long term solution given the fluctuation of gas prices over time. The proposed opcode `0x46 MEDGASPRICE` would return the median gas price for the parent block. This would allow the implementation of upper bounds on transaction gas prices as a function of the last block median gas price.
+
+## Motivation
+
+With the emergence of rollups as core mechanisms in scaling Ethereum there are a number of common transactions that can be front-run. Optimistic rollups rely on the submission of fraud proofs to maintain the integrity of their systems. As a result actors submitting fraud proofs typically receive a financial reward for doing so. This opens a trivial front-running strategy of watching the mempool for fraud proof submissions and copying such transactions with a much higher gas price to reap the reward. Such front-runners do not perform validation independently and de-incentivize others from performing validation. Adding a mechanism enforcing an upper bound on gas prices for a transaction could be an effective defense against such front-running attacks.
+
+## Specification
+
+If `block.number >= TBD`, add a new opcode:
+
+MEDGASPRICE: 0x46
+
+Pushes the median gas price of the parent block onto the stack.<br />
+Gas costs: 8
+
+The name `MEDGASPRICE` was chosen because the median gas price of the network can only be calculated from the latest complete block. Thus transactions being executed should expect the median gas price to be calculated from the previous block.
```

Comment:
This should be under rationale if included, but seems rather self-explanatory.

Reply:
I think it's somewhat counterintuitive for the code `block.medgasprice` to refer to the previous block gas price. I kept it in but moved it to the rationale section.